prov/efa/src/efa_domain.c

```diff
@@ -351,6 +354,20 @@ static int efa_domain_close(fid_t fid)
 		efa_domain->cache = NULL;
 	}
 
+	/* Clean up ah_map if any entries remain */
+	ofi_genlock_lock(&efa_domain->util_domain.lock);
+	if (efa_domain->ah_map) {
+		EFA_WARN(FI_LOG_DOMAIN, "AH map not empty during domain close! Cleaning up ...\n");
```

Comment:
```
ret = ofi_domain_close(&efa_domain->util_domain);
```
Should return -FI_EBUSY if users have domain resources open (AV, EP....).  We should move deleting the util domain to the top of this function.  Also, this warning should only be possible to hit if we have a bug in our code.  Warnings, are for users, asserts() are for us, and this should be an assert.  The rest of the cleanup should also be removed from here. I recommend a follow up PR that fixes our  domain cleanup logic.